radis/lbl/base.py

```diff
@@ -1596,20 +1714,45 @@ def calc_weighted_trans_moment(self):
             # still much faster than the groupby().apply() method (see radis<=0.9.19)
             # (tested + see https://stackoverflow.com/questions/44954514/efficient-way-to-conditionally-populate-elements-in-a-pandas-groupby-object-pos)
 
-            dgb = df.groupby(by=["id", "iso"])
-            for (id, iso), idx in dgb.indices.items():
-                molecule = get_molecule(id)
+            if "iso" in df:
+                isotope_set = df.iso.unique()
+                isotope = isotope_set[0]
+
+            else:
+                isotope = df.attrs["iso"]
+
+            molecule = get_molecule(id)
+
+            if ("iso" in df and len(isotope_set) != 1) or ("iso" not in df and isotope):
+                dgb = df.groupby(by=["iso"])
+                for (iso), idx in dgb.indices.items():
+                    state = self.input.state
+                    parsum = self.get_partition_function_calculator(
+                        molecule, iso, state
+                    )  # partition function
+                    df.at[idx, "Qref"] = parsum.at(Tref, update_populations=False)
+                    # ... note: do not update the populations here, so populations in the
+                    # ... energy level list correspond to the one calculated for T and not Tref
+
+                    if radis.DEBUG_MODE:
+                        if "id" in df:
+                            assert (df.loc[idx, "id"] == id).all()
+                        assert (df.loc[idx, "iso"] == iso).all()
+
+            else:
                 state = self.input.state
                 parsum = self.get_partition_function_calculator(
-                    molecule, iso, state
+                    molecule, isotope_set[0], state
                 )  # partition function
                 df.at[idx, "Qref"] = parsum.at(Tref, update_populations=False)
```

Comment:
is `idx` defined here?  (we're not in the groupby() case. Why didn't it trigger a syntax error ? )

Reply:
The entire else block which contained this line was actually redundant. All the cases already seemed to be handled prior to this. So the lines that followed were a miss. So, it was not giving any `Variable used before declaration error`. 

As for the fix in L555. `get_Evib_CDSD_pc_1iso` is called in L562 and we are already passing idx in there. So we do not require this a `try-except` here.